aten/src/ATen/native/cuda/UpSampleBilinear2d.cu

```diff
@@ -1,25 +1,312 @@
+// Adapted from interp.cpp from Caffe util by Pauline Luc
+// Originally developed by George Papandreou
 #include <ATen/ATen.h>
+#include <ATen/AccumulateType.h>
 #include <ATen/NativeFunctions.h>
-#include <ATen/LegacyTHFunctions.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/Utils.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+#include <ATen/native/cuda/UpSample.cuh>
 
 namespace at {
 namespace native {
+namespace {
+
+template <typename scalar_t, typename accscalar_t>
+#ifdef __HIP_PLATFORM_HCC__
+C10_LAUNCH_BOUNDS_1(1024)
```

Comment:
To be applied unconditionally, after fixing kernel launch site!

Reply:
@ezyang I have applied the changes pointed in this PR, just not sure now what approach I should take to fix the kernel launch site. do you have any recommendation here?

Reply:
At a high level, what you need to do is make sure you never launch the kernel with bounds greater than the bounds you hard-code here (1024). More concretely, you should change the launch code to stop pulling out the limit from the configuration, and just configure the bounds themselves. I'd recommend reading some references about how to do cuda launches. If you're still confused, shout and I can figure it out (this is also my first time working with this code.)

Reply:
The limit still has to be pulled from config, and then a min taken on it (there's no guarantee some future hardware won't have a limit of 512 instead of 1024).

Reply:
well I've investigated more about this topic, but I didn't find a good answer. 
I just changed the maxthread to `at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock / 2` and locally it passed. if you have any other idea to address that or any reference to read, that would be very helpful :)

Reply:
Sorry about not responding to this earlier.  I don't think dividing my two is the correct thing to do here; it doesn't seem justified by the semantics of the parameters involved.

What Natalia wrote in her comment is exactly what should be done. The max thread should be `std::min(at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, 1024)`, where `1024` is the number that was given to the `LAUNCH_BOUNDS` annotation. This prevents us attempting to launch the kernel with more threads than the device supports (the first argument), or 1024 (the second argument.)

Your change probably results in a launch bounds which is less than either of these limits, but it probably reduces the parallelism that is possible here.

Reply:
oh I see ... I understand it now ... thanks for the explanation!